test/e2e/minscale_readiness_test.go

```diff
@@ -85,24 +85,32 @@ func TestMinScale(t *testing.T) {
 	revName := latestRevisionName(t, clients, names.Config, "")
 	serviceName := PrivateServiceName(t, clients, revName)
 
-	t.Log("Waiting for revision to scale to minScale before becoming ready")
-	if lr, err := waitForDesiredScale(clients, serviceName, gte(minScale)); err != nil {
-		t.Fatalf("The revision %q scaled to %d < %d before becoming ready: %v", revName, lr, minScale, err)
-	}
-
-	t.Log("Waiting for revision to become ready")
+	t.Log("Waiting for new revision to become ready")
 	if err := v1test.WaitForRevisionState(
 		clients.ServingClient, revName, v1test.IsRevisionReady, "RevisionIsReady",
 	); err != nil {
 		t.Fatalf("The Revision %q did not become ready: %v", revName, err)
 	}
 
+	revision, err := clients.ServingClient.Revisions.Get(context.Background(), revName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("An error occurred getting revision %v, %v", revName, err)
+	}
+	if replicas := *revision.Status.ActualReplicas; replicas < minScale {
+		t.Fatalf("Container is indicated as healthy. Expected actual replicas for revision %v to be %v but got %v", revision.Name, minScale, replicas)
+	}
```

Comment:
I'm hitting this condition when running locally against a broken serving release. I wouldn't expect that so it makes me think there's a delay when ActualReplicas is updated. 

This fails sometimes when scaling up the first revision or second. Because of that I don't think we can reliably use this value in the test. 

This is probably a separate issue to the one you're addressing

Reply:
This check is also duplicated on line 117